src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -0,0 +1,840 @@
+			if (!IsPressed(args.Pointer))
+			{
+				handledInManaged |= ReleaseCapture(args);
+			}
+
+			return handledInManaged;
+		}
+
+		/// <summary>
+		/// When the system cancel a pointer pressed, either
+		/// 1. because the pointing device was lost/disconnected,
+		/// 2. or the system detected something meaning full and will handle this pointer internally.
+		/// This second case is the more common (e.g. ScrollViewer) and should be indicated using the <paramref name="isSwallowedBySystem"/> flag.
+		/// </summary>
+		/// <param name="isSwallowedBySystem">Indicates that the pointer was muted by the system which will handle it internally.</param>
+		private bool OnNativePointerCancel(PointerRoutedEventArgs args, bool isSwallowedBySystem)
+		{
+			var isIrrelevant = ValidateAndUpdateCapture(args); // Check this *before* updating the pressed / over states!
+
+			// When a pointer is cancelled / swallowed by the system, we don't even receive "Released" nor "Exited"
+			SetPressed(args, false, muteEvent: true);
+			SetOver(args, false, muteEvent: true);
+
+			if (isIrrelevant)
+			{
+				return false;
+			}
+		
+			if (_gestures.IsValueCreated)
+			{
+				_gestures.Value.CompleteGesture();
+			}
+
+			var handledInManaged = false;
+			if (isSwallowedBySystem)
+			{
+				handledInManaged |= ReleaseCapture(args, forceCaptureLostEvent: true);
+			}
+			else
+			{
+				args.Handled = false;
+				handledInManaged |= RaisePointerEvent(PointerCanceledEvent, args);
+				handledInManaged |= ReleaseCapture(args);
+			}
+
+			return handledInManaged;
+		}
+
+		private static (UIElement sender, RoutedEvent @event, PointerRoutedEventArgs args) _pendingRaisedEvent;
+		private bool RaisePointerEvent(RoutedEvent evt, PointerRoutedEventArgs args)
+		{
+			try
+			{
+				_pendingRaisedEvent = (this, evt, args);
+				return RaiseEvent(evt, args);
+			}
+			catch (Exception e)
+			{
+				if (this.Log().IsEnabled(LogLevel.Error))
+				{
+					this.Log().Error($"Failed to raise '{evt.Name}': {e}");
+				}
+
+				return false;
+			}
+			finally
+			{
+				_pendingRaisedEvent = (null, null, null);
+			}
+		}
+		#endregion
+
+		#region Pointer over state (Updated by the partial API OnNative***)
+		/// <summary>
+		/// Indicates if a pointer (no matter the pointer) is currently over the element (i.e. OverState)
+		/// WARNING: This might not be maintained for all controls, cf. remarks.
+		/// </summary>
+		/// <remarks>
+		/// This flag is updated by the managed Pointers events handling, however on Android and WebAssembly,
+		/// pointer events are marshaled to the managed code only if there is some listeners to the events.
+		/// So it means that this flag will be maintained only if you subscribe at least to one pointer event
+		/// (or override one of the OnPointer*** methods).
+		/// </remarks>
+		internal bool IsPointerOver { get; set; } // TODO: 'Set' should be private, but we need to update all controls that are setting
```

Comment:
same for these TODOs in this file as well

Reply:
This is more a comment for future development than a real to-do. I'll keep it like that.